cpu/fe310/include/periph_cpu.h

```diff
@@ -94,6 +94,14 @@ typedef struct {
  */
 #define PERIPH_TIMER_PROVIDES_SET
 
+/**
+ * @name    Power management configuration
+ * @{
+ */
+#define PROVIDES_PM_OFF
```

Comment:
Instead of defining this, I would rather remove the specific implementation of pm_off for this CPU: it's empty and probably doesn't do what it should.

Reply:
My assumption was that the reason behind this is that the default `pm_off` implementation does not work on FE310. Is that assumption incorrect?

Reply:
> Is that assumption incorrect?

Good question. I'll test that and report here.

Reply:
So I tested with `tests/periph_pm`. Results: 

- on master (which provides a specific and empty implemetation of pm_off):

<details>

```
Bench Clock Reset Complete
2020-01-30 07:51:13,535 # 
2020-01-30 07:51:13,646 # ATE0-->ATE0
2020-01-30 07:51:13,662 # OK
2020-01-30 07:51:13,819 # AT+BLEINIT=0-->OK
2020-01-30 07:51:13,975 # AT+CWMODE=0-->OK
2020-01-30 07:51:13,976 # 
2020-01-30 07:51:13,986 # main(): This is RIOT! (Version: 2020.04-devel-150-g0d318)
2020-01-30 07:51:13,992 # This application allows you to test the CPU power management.
2020-01-30 07:51:13,997 # Layered support is not unavailable for this CPU. Reset the CPU if
2020-01-30 07:51:13,998 # needed.
> off
2020-01-30 07:51:16,641 #  off
2020-01-30 07:51:16,642 # CPU will turn off.
> 2020-01-30 07:51:17,281 #  off
2020-01-30 07:51:17,283 # CPU will turn off.
> 2020-01-30 07:51:17,445 #  off
2020-01-30 07:51:17,447 # CPU will turn off.
> 2020-01-30 07:51:17,601 #  off
2020-01-30 07:51:17,602 # CPU will turn off.
> 2020-01-30 07:51:17,721 #  off
2020-01-30 07:51:17,722 # CPU will turn off.
> 2020-01-30 07:51:18,639 # Exiting Pyterm
```

</details>

=> it does nothing, so the shell is still active although it says "CPU will turn off.".

- With this diff applied to master:

```diff
diff --git a/cpu/fe310/include/periph_cpu.h b/cpu/fe310/include/periph_cpu.h
index 6c44f875c6f..1449988d734 100644
--- a/cpu/fe310/include/periph_cpu.h
+++ b/cpu/fe310/include/periph_cpu.h
@@ -27,6 +27,8 @@
 extern "C" {
 #endif
 
+#define PROVIDES_PM_SET_LOWEST
+
 /**
  * @brief   Length of the CPU_ID in octets
  */
diff --git a/cpu/fe310/periph/pm.c b/cpu/fe310/periph/pm.c
index 0e49f68694d..6576820f7ff 100644
--- a/cpu/fe310/periph/pm.c
+++ b/cpu/fe310/periph/pm.c
@@ -26,10 +26,6 @@ void pm_set_lowest(void)
     __asm__ volatile ("wfi");
 }
 
-void pm_off(void)
-{
-}
-
 void pm_reboot(void)
 {
     AON_REG(AON_WDOGKEY) = AON_WDOGKEY_VALUE;
```

<details>

```
Bench Clock Reset Complete
2020-01-30 07:50:40,359 # 
2020-01-30 07:50:40,456 # ATE0-->ATE0
2020-01-30 07:50:40,473 # OK
2020-01-30 07:50:40,629 # AT+BLEINIT=0-->OK
2020-01-30 07:50:40,785 # AT+CWMODE=0-->OK
2020-01-30 07:50:40,786 # 
2020-01-30 07:50:40,797 # main(): This is RIOT! (Version: 2020.04-devel-150-g0d318)
2020-01-30 07:50:40,802 # This application allows you to test the CPU power management.
2020-01-30 07:50:40,808 # Layered support is not unavailable for this CPU. Reset the CPU if
2020-01-30 07:50:40,808 # needed.
> help
2020-01-30 07:50:49,366 #  help
2020-01-30 07:50:49,369 # Command              Description
2020-01-30 07:50:49,373 # ---------------------------------------
2020-01-30 07:50:49,375 # off                  turn off
2020-01-30 07:50:49,378 # reboot               reboot
> off
2020-01-30 07:50:52,153 #  off
2020-01-30 07:50:52,154 # CPU will turn off.
help
mskfdmkf
sfkjsdklfj
lskdfj
2020-01-30 07:51:01,143 # Exiting Pyterm
```

</details>

=> The CPU gives the impression to be off. So this is already much better.

Reply:
Should I just apply your patch then?

Reply:
Just adapt this PR, that will be perfectly fine.

Reply:
Done, thanks for your detailed tests!